go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -1184,3 +1185,146 @@ func (bnc *BaseNetworkController) GetSamplingConfig() *libovsdbops.SamplingConfi
 }
+
+const advertisedUDNIsolationACLID = "advertised-udn-isolation"
+const advertisedUDNSubnetsAddressSet = "advertised-udn-subnets"
+const advertisedUDNSubnetsCtrl = "advertised-udn-subnets-controller"
+
+func GetAdvertisedUDNSubnetsAddressSetDBIDs() *libovsdbops.DbObjectIDs {
+	return libovsdbops.NewDbObjectIDs(libovsdbops.AddressSetAdvertisedUDN, advertisedUDNSubnetsCtrl, map[libovsdbops.ExternalIDKey]string{
+		libovsdbops.ObjectNameKey: advertisedUDNSubnetsAddressSet,
+	})
+}
+
+func GetAdvertisedUDNSubnetsDropACLdbIDs() *libovsdbops.DbObjectIDs {
+	return libovsdbops.NewDbObjectIDs(libovsdbops.ACLAdvertisedUDN, advertisedUDNIsolationACLID,
+		map[libovsdbops.ExternalIDKey]string{
+			libovsdbops.ObjectNameKey: advertisedUDNSubnetsCtrl,
+		})
+}
+
+func GetAdvertisedUDNSubnetsAcceptACLdbIDs(networkName string) *libovsdbops.DbObjectIDs {
+	return libovsdbops.NewDbObjectIDs(libovsdbops.ACLAdvertisedUDN, advertisedUDNIsolationACLID,
+		map[libovsdbops.ExternalIDKey]string{
+			libovsdbops.ObjectNameKey: networkName,
+		})
+}
+
+func BuildAdvertisedUDNSubnetsDropACL(advertisedUDNSubnetsAddressSet addressset.AddressSet) *nbdb.ACL {
+	var dropMatches []string
+	v4AddrSet, v6AddrSet := advertisedUDNSubnetsAddressSet.GetASHashNames()
+	if v4AddrSet != "" {
+		dropMatches = append(dropMatches, fmt.Sprintf("(ip4.src == $%s && ip4.dst == $%s)", v4AddrSet, v4AddrSet))
+	}
+	if v6AddrSet != "" {
+		dropMatches = append(dropMatches, fmt.Sprintf("(ip6.src == $%s && ip6.dst == $%s)", v6AddrSet, v6AddrSet))
+	}
+
+	dropACL := libovsdbutil.BuildACL(
+		GetAdvertisedUDNSubnetsDropACLdbIDs(),
+		types.DefaultDenyPriority,
+		strings.Join(dropMatches, " || "),
+		nbdb.ACLActionDrop,
+		nil,
+		libovsdbutil.LportEgress)
+	dropACL.Tier = types.PrimaryACLTier
+	return dropACL
+}
+
+func (bnc *BaseNetworkController) addAdvertisedUDNIsolation(nodeName string) error {
+	var acceptMatches, cidrs []string
+	var ops []ovsdb.Operation
+
+	addrSet, err := bnc.addressSetFactory.GetAddressSet(GetAdvertisedUDNSubnetsAddressSetDBIDs())
+	if err != nil {
+		return err
+	}
+
+	for _, subnet := range bnc.Subnets() {
+		ipPrefix := "ip4"
+		if utilnet.IsIPv6CIDR(subnet.CIDR) {
+			ipPrefix = "ip6"
+		}
+		acceptMatches = append(acceptMatches, fmt.Sprintf("(%s.src == %s && %s.dst == %s)", ipPrefix, subnet.CIDR, ipPrefix, subnet.CIDR))
+		cidrs = append(cidrs, subnet.CIDR.String())
+
+	}
+
+	addrOps, err := addrSet.AddAddressesReturnOps(cidrs)
+	if err != nil {
+		return fmt.Errorf("failed to add addresses %q to the address set: %w", cidrs, err)
+	}
+	ops = append(ops, addrOps...)
+
+	if len(acceptMatches) > 0 {
+		acceptACL := libovsdbutil.BuildACL(
+			GetAdvertisedUDNSubnetsAcceptACLdbIDs(bnc.GetNetworkName()),
+			types.DefaultAllowPriority,
+			strings.Join(acceptMatches, " || "),
+			nbdb.ACLActionPass,
+			nil,
+			libovsdbutil.LportEgress)
+		acceptACL.Tier = types.PrimaryACLTier
+
+		ops, err = libovsdbops.CreateOrUpdateACLsOps(bnc.nbClient, ops, nil, acceptACL)
```

Comment:
so this is going to be 1 pass ACL per UDN right?
not 1 gigantic one for all UDNs?

Reply:
yup, added some better docs to functions.